syntax = "proto3";

package text2video;

// Request message
message TextPrompt {
  string prompt = 1;
}

// Response message
message VideoResponse {
  string video_path = 1;
  string message = 2;
  int32 status_code = 3;
}

// Service definition
service VideoGenerator {
  rpc Generate (TextPrompt) returns (VideoResponse);
}
